### PR TITLE
TST: Reorganize only the test_environment.py tests into Integration/Unit Tests

### PR DESCRIPTION
**Pull request by lucasfourier**
## Pull request type

- [ ] Code changes (bugfix, features)
- [X] Code maintenance (refactoring, formatting, tests)
- [ ] ReadMe, Docs and GitHub updates
- [ ] Other (please describe):

## Checklist

- [ ] Tests for the changes have been added (if needed)
- [ ] Docs have been reviewed and added / updated
- [X] Lint (`black rocketpy/ tests/`) has passed locally 
- [X] All tests (`pytest tests -m slow --runslow`) have passed locally
- [ ] `CHANGELOG.md` has been updated (if relevant)

## Current behavior

Currently, the folder tests/integration is empty. 

## New behavior

Integration tests folder is no longer empty and contains integration tests. They are not new, they were simply moved to the right place.

## Breaking change

- [ ] Yes
- [X] No

## Additional information

image

### CODE REVIEW

**Comment by Gui-FernandesBR** [comment]
@lucasfourier , why moving just some of the tests of the tests from `tests/test_environment.py` instead of all of them?
What does set these tests apart?

**Comment by lucasfourier** [comment]
Because the tests that have been moved can be defined as integration tests as opposed to the remaining ones. The fundamental trait is the communication with external dependencies.

**Comment by Gui-FernandesBR** [comment]
@lucasfourier Can you please re-work on this branch to move all the integration tests at once instead of just some of the tests?

File: tests/unit/__init__.py
Comment (Gui-FernandesBR) [inline tests/unit/__init__.py:1]: @lucasfourier can you use this `__init__.py` file to define what a unit test is? Use a top-file docstring to include a short, but accurate description of what should or should not be included in the `tests.unit` module

Same thing is valid for the new integration module.

**Comment by Gui-FernandesBR** [comment]
@lucasfourier what is the reason for these methods be considered integration and not unit tests? They were moved from unit tests to the integration tests.

1. `test_environment_export_environment_exports_valid_environment_json`
2. `test_date_aware_set_date_saves_custom_timezone`
3. `test_date_naive_set_date_saves_utc_timezone_by_default`

The definition of unit and integration *are not clear enough*. I think it is important that you write down these definition before proceeding with any changes in this PR. This should take a few minutes, but save you from questions like the 3 above.

**Comment by lucasfourier** [comment]
Tests with connection to API'S/interfaces were moved into the integration folder and therefore are only run with --runslow flag, just for the record.

**Review by Gui-FernandesBR — APPROVED** [review]
Good work! @lucasfourier

**Pull request merged**